docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+
+Alternatively, you can set a very permissive constraint and let composer find a SilverStripe 4 compatible version. After you're done updating your dependencies, make sure you adjust your constraints to be more specific.
+
+Once you've updated your third-party packages constraints, try updating your dependencies by running `composer update`. If composer can't find all your dependencies it will throw an error.
+
+### Resolving conflicts
+Whatever you have updated your dependencies with the upgrader or manually, you'll likely have some conflicts to resolve. Running a `composer update` will tell you which package are conflicted and suggested alternative combinations of package that might work.
+
+The most typical reason for a conflict is that the maintainer of a package has not release a version compatible with SilverStripe 4.
+
+If the maintainer of the package is in the process of upgrading to SilverStripe 4, a development version of the package might be available. In some cases, it can be worthwhile to look up the repository of the package or to reach out to the maintainer.
+
+To resolve a conflict you can either:
+* remove the package from your project, if it is not essential
+* fork the package and maintain it yourself.
+
+If you decide to fork a package, remove it from your `composer.json` file and from your `.gitignore` file. Then track the package's codebase in your project source control. Alternatively, you can track the fork as a private  package if you do not which to merge it in your code.
+
+<div class="info" markdown="1">
+If you're taking the time to upgrade a third party package, consider doing a pull request against the original project so other developers can benefit from your work.
+</div>
+
+### Finalising your dependency upgrade
+Once you've resolved all conflicts in your composer.json file, `composer update` will be able to run without errors.
+
+This will install your new dependencies. You'll notice many of the folders in the root of your project will disappear. That's because SilverStripe 4 packages can be installed in the vendor folder like generic PHP packages.
+
+If you've decided to use recipes, some generic files will be copied from the recipe into your project. The `extra` attribute in your `composer.json` file will be updated to keep track of those new files.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.  
+
+
+## Step 2 - Update your environment configuration {#env}{#step2}
+
+The php configuration `_ss_environment.php` file has been replaced in favour of a non-executable
+`.env` file, which follows a syntax similar to a `.ini` file for key/value pair assignment. Your `.env` file may be placed in either the web root, or your project root.
```

Comment:
I think I got it from the changelog https://docs.silverstripe.org/en/4/changelogs/4.0.0/#env

> Like the old php file, .env may be placed in either the web root, or one level above.

I remember being confused when I came across this part as well. I took it to say that your `.env` file can be in your web root or your project root. But since this is specific to 4.0 the web root and your project root will always be the same.

I changed it to say:
> Your `.env` file may be placed in your project root, or one level above your project root.